sklearn/isotonic.py

```diff
@@ -391,8 +399,15 @@ def __setstate__(self, state):
         We need to rebuild the interpolation function.
         """
         super().__setstate__(state)
+        # for backward compatibility with objects pickled with versions
+        # strictly older than 0.23.
```

Comment:
We don't support backward compat of pickles between versions.

I won't fight too hard if you think we should keep @ogrisel, but IMHO we have enough on our plate with public attributes already, so I don't think it's worth bothering about private ones on top of that

Reply:
Or if we keep this, maybe we should deprecate the attributes for real. Else, this code will stay there forever.

Reply:
I agree with @NicolasHug here with not supporting pickling from a maintenance point of view.

Reply:
Ok, it's just that I thought it would not be complex to maintain for this specific case. Let me remove it to keep the code as simple as possible.